src/raft.rs

```diff
@@ -389,6 +389,20 @@ impl<T: Storage> Raft<T> {
         self.batch_append = batch_append;
     }
 
+    /// Uses a new quorum function.
+    pub fn set_quorum(&mut self, quorum_fn: fn(usize) -> usize) {
+        self.quorum_fn = quorum_fn;
+        match self.state {
+            StateRole::Leader => {
+                if self.maybe_commit() {
```

Comment:
Is it necessary to check pending reads?

Reply:
It's an existing bug. Not only should check when quorum_fn is changed, but also should check when a voter is removed.